homeassistant/components/emoncms/config_flow.py

```diff
@@ -94,6 +94,17 @@ async def async_step_user(
                 self.include_only_feeds = user_input.get(CONF_ONLY_INCLUDE_FEEDID)
                 self.url = user_input[CONF_URL]
                 self.api_key = user_input[CONF_API_KEY]
+                emoncms_client = EmoncmsClient(
+                    self.url, self.api_key, session=async_get_clientsession(self.hass)
+                )
+                emoncms_unique_id = await emoncms_client.async_get_uuid()
```

Comment:
I think you should use set_unique_id + abort_uniqueid_match

As it stands you never store it on creation...

Reply:
I store it in the sensor module
https://github.com/alexandrecuer/core/blob/21ca3492d5764789dd66c21d775e0bb4028fcca0/homeassistant/components/emoncms/sensor.py#L170

Reply:
Yes, but that is extra...
I think it should be using a config entry migration

Reply:
I think I've got it and used async_set_unique_id as you suggested. Could not find anything like abort_unique_id_match....there is a abort_if_unique_id_mismatch method but this is not what I want. So I've kept the async_abort

Reply:
OK I used also self._abort_if_unique_id_configured() so now it must be what you are speaking about ie : set_unique_id + abort_unique_id_match